test/mechanics/integrands/NeumannBc.cpp

```diff
@@ -80,324 +80,61 @@ BOOST_AUTO_TEST_CASE(NeumannBc1Din2D)
-                nodePtrs.push_back(&nodes[3]);
-
-                nodePtrs.push_back(&nodes[5]);
-                nodePtrs.push_back(&nodes[8]);
-                nodePtrs.push_back(&nodes[9]);
-
-                nodesDispPtrs.push_back(&nodesDisp[1]);
-                nodesDispPtrs.push_back(&nodesDisp[2]);
-                nodesDispPtrs.push_back(&nodesDisp[3]);
-
-                nodesDispPtrs.push_back(&nodesDisp[5]);
-                nodesDispPtrs.push_back(&nodesDisp[8]);
-                nodesDispPtrs.push_back(&nodesDisp[9]);
-
-                transform3d2d.block(0 * 3, 1 * 3, 3, 3) = Eigen::Matrix<double, 3, 3>::Identity();
-                transform3d2d.block(1 * 3, 2 * 3, 3, 3) = Eigen::Matrix<double, 3, 3>::Identity();
-                transform3d2d.block(2 * 3, 3 * 3, 3, 3) = Eigen::Matrix<double, 3, 3>::Identity();
-
-                transform3d2d.block(3 * 3, 5 * 3, 3, 3) = Eigen::Matrix<double, 3, 3>::Identity();
-                transform3d2d.block(4 * 3, 8 * 3, 3, 3) = Eigen::Matrix<double, 3, 3>::Identity();
-                transform3d2d.block(5 * 3, 9 * 3, 3, 3) = Eigen::Matrix<double, 3, 3>::Identity();
-                break;
-            }
-            }
-
-            // coordinate element
-            InterpolationTriangleQuadratic coordinateInterpolation(3);
-            ElementCollectionFem element({nodePtrs, coordinateInterpolation});
-
-            // displacement nodes
-            DofType dof("displacements", 3);
-            InterpolationTriangleQuadratic displacementInterpolation(3);
-            element.AddDofElement(dof, {nodesDispPtrs, displacementInterpolation});
-
-            Eigen::Vector3d p(1, 1, 1);
-
-            Integrands::TimeDependent::NeumannBc<3> neumannIntegrand(dof, p);
-
-            CellData cellData(element);
-            Jacobian dummyJac(
-                    element.CoordinateElement().ExtractNodeValues(), // jacobian not needed for N.
-                    element.CoordinateElement().GetDerivativeShapeFunctions(Eigen::VectorXd::Constant(2, 1, 0.)), 3);
-
-            CellIpData cellIpData(element, dummyJac, surfaceIP);
-
-            Eigen::VectorXd TetrahedronDN =
-                    NuTo::ShapeFunctions3D::ShapeFunctionsTetrahedronOrder2(volumetricParameters);
-
-            Eigen::MatrixXd NTetrahedron = NuTo::Matrix::N(TetrahedronDN, 10, 3);
-            auto gradient = neumannIntegrand.Gradient(cellData, cellIpData);
-            BoostUnitTest::CheckEigenMatrix(gradient[dof], transform3d2d * NTetrahedron.transpose() * p);
-        }
+        CellIpData cellIpData(element, dummyJac, points[ip]);
+
+
+        auto gradient = neumannIntegrand.Gradient(cellData, cellIpData);
+
+        Eigen::VectorXd referenceResultOnIP = (1. / 6.) * Eigen::VectorXd::Ones(9);
+        referenceResultOnIP.block(3 * ip, 0, 3, 1) = (2. / 3.) * Eigen::VectorXd::Ones(3);
+
+        // results on each ip
+        BoostUnitTest::CheckEigenMatrix(gradient[dof], referenceResultOnIP);
```

Comment:
My idea was to actually integrate the neumann integrand over the cell and compare the integrated result with something like `p * area(element)`. This would require the jacobian which is already tested in `Jacobian.cpp`. So I really like this test, since it is independent from the jacobian and it is not tested twice.

But the values of `refenceResultOnIP` are not obvious to me. Can you comment on where these values come from? Or what do you think about mixing both testing approaches? Actually perform the integration with a _prescribed_ jacobi determinant. So in this case the area should be `sqrt(3)/2`. The resulting forces are just `sqrt(3)/2 * ones`. So the sum of all x components should be `sqrt(3)/2`, same for y and z.